### ISSUE TITLE
SC2154 triggers incorrectly on ${!prefix@} expansions

### ISSUE THREAD

**Issue by ribbons** [koalaman/shellcheck#1500]
- Rule Id: SC2154
- My shellcheck version: 0.6.0
- [x] The rule's wiki page does not already cover this at https://shellcheck.net/wiki/SC2154
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

This is a very similar issue to #443, just with `@` rather than `*`

#### Here's a snippet that shows the problem:

```sh
#!/bin/bash

# shellcheck disable=SC2034
osfoo=osfoov osbar=osbarv osblah=osblahv

echo "${!os@}"
```

#### Here's what shellcheck currently says:

```
Line 6:
echo "${!os@}"
      ^-- SC2154: os is referenced but not assigned.
```

#### Here's what I wanted or expected to see:

No warnings

**Comment by harrylee2015** [comment]
I also encountered this problem, how to solve it?
```bash
In ./build/docker-compose-down.sh line 36:
    num=${#remains[@]}
        ^-- SC2154: remains is referenced but not assigned.
```

**Comment by harrylee2015** [comment]
```bash
# shellcheck disable=SC2154
```

**Comment by balanco** [comment]
This is a kind reminder that the flagged '${!variable...**@**}' issue persists as of shellcheck v0.7.1 (as opposed to the fixed '${!variable...*****}' #issue  443).
A different use case is the ensuring that any variables starting with a common name prefix are unset
(say, temporary variables with names starting with 'tmp_', i.e.: 'tmp_bla1', 'tmp_bla2', etc.):

`unset "${!tmp_@}"        # note: the builtin 'unset' does not fail if no argument is supplied`

is likewise incorrectly diagnosed (if "just" as a warning): 

`SC2154 tmp_ is referenced but not assigned.`

(the workaround of explicitly shellcheck-commenting the code line of course works, but.)

**Comment by TinCanTech** [comment]
Try `v0.72` or `https://www.shellcheck.net/`

**Comment by mminot-yseop** [comment]
> Try `v0.72` or `https://www.shellcheck.net/`

Seems OK on the website indeed. I guess this issue could be closed, then? 🤔
